twine/settings.py

```diff
@@ -338,6 +340,21 @@ def check_repository_url(self) -> None:
                 repository_url, utils.DEFAULT_REPOSITORY, utils.TEST_REPOSITORY
             )
 
+        if "pypi.org" in repository_url and repository_url not in [
+            utils.DEFAULT_REPOSITORY,
+            utils.TEST_REPOSITORY,
+        ]:
+            raise exceptions.InvalidPyPIUploadURL(
+                f"The configured repository {repository_url} is not a known "
+                f"PyPI repository.\n"
+                f"You probably want {utils.DEFAULT_REPOSITORY} "
+                f"or {utils.TEST_REPOSITORY}.\n"
+                f"Check your --repository-url value "
+                f"or the repository configuration in {self.config_file}.\n"
+                f"See https://packaging.python.org/specifications/pypirc/ "
+                f"for more details."
+            )
```

Comment:
Following up on https://github.com/pypa/twine/pull/809#discussion_r706836618, I generalized the logic to catch more cases, and tweaked the message. Here's what it looks like:

```
% twine upload --repository-url https://upload.pypi.org/legacy dist/*
InvalidPyPIUploadURL: The configured repository https://upload.pypi.org/legacy is not a known PyPI repository.
You probably want https://upload.pypi.org/legacy/ or https://test.pypi.org/legacy/.
Check your --repository-url value or the repository configuration in ~/.pypirc.
See https://packaging.python.org/specifications/pypirc/ for more details.
```

In the process, I noted that the logic and messages is very similar to [`twine.utils.check_status_code`](https://github.com/pypa/twine/blob/f54311486f1a27e732fa0118ac1b2b71f1c32c10/twine/utils.py#L172-L195). I'd really like to refactor that, maybe by moving the messages into the exception definitions. But I'm guessing that should be a follow-up PR, related to https://github.com/pypa/twine/issues/586.